go-controller/pkg/ovn/policy.go

```diff
@@ -1133,53 +1181,66 @@ func (oc *Controller) destroyNetworkPolicy(np *networkPolicy, nsInfo *namespaceI
 		return true
 	})
 
+	ingressPGName := defaultDenyPortGroup(np.namespace, ingressDefaultDenySuffix)
+	egressPGName := defaultDenyPortGroup(np.namespace, egressDefaultDenySuffix)
+
 	ingressDenyPorts, egressDenyPorts := oc.localPodDelDefaultDeny(np, ports...)
 
 	ops := []ovsdb.Operation{}
 	var err error
-	if len(nsInfo.networkPolicies) == 0 {
-		ops, err = libovsdbops.DeletePortGroupsOps(oc.nbClient, ops, nsInfo.portGroupIngressDenyName)
+	// we haven't deleted our np from the namespace yet so there should be 1 policy
+	// if there are no more policies left on the namespace
+	if nsInfo != nil && len(nsInfo.networkPolicies) == 1 {
+		ops, err = libovsdbops.DeletePortGroupsOps(oc.nbClient, ops, ingressPGName)
 		if err != nil {
-			klog.Errorf("%v", err)
+			return fmt.Errorf("failed to make ops for delete ingress port group: %s, for policy: %s/%s, error: %v",
+				ingressPGName, np.namespace, np.name, err)
 		}
-		ops, err = libovsdbops.DeletePortGroupsOps(oc.nbClient, ops, nsInfo.portGroupEgressDenyName)
+		ops, err = libovsdbops.DeletePortGroupsOps(oc.nbClient, ops, egressPGName)
 		if err != nil {
-			klog.Errorf("%v", err)
+			return fmt.Errorf("failed to make ops for delete egress port group: %s, for policy: %s/%s, error: %v",
+				egressPGName, np.namespace, np.name, err)
 		}
 	} else {
-		ops, err = libovsdbops.DeletePortsFromPortGroupOps(oc.nbClient, ops, nsInfo.portGroupIngressDenyName, ingressDenyPorts...)
+		ops, err = libovsdbops.DeletePortsFromPortGroupOps(oc.nbClient, ops, ingressPGName, ingressDenyPorts...)
 		if err != nil {
-			klog.Errorf(err.Error())
+			return fmt.Errorf("failed to make ops for ingress port group: %s, for policy: %s/%s, to remove ports: %#v,"+
+				" error: %v", ingressPGName, np.namespace, np.name, ingressDenyPorts, err)
 		}
 		ops, err = libovsdbops.DeletePortsFromPortGroupOps(oc.nbClient, ops, nsInfo.portGroupEgressDenyName, egressDenyPorts...)
```

Comment:
And if the only place we're using nsInfo is in the "if # policies = 1" block, could we just have the caller pass `len(nsInfo.networkPolicies)` as a "lastPolicy bool" argument instead of nsInfo itself?

Reply:
oops I missed this one, yeah should be egressPGName, and yeah the suggestion to use a bool is better. Will do that.